### DISCUSSION TITLE
s2_m4 missing Module library

### DISCUSSION THREAD

**Discussion by math322** [emsesp/EMS-ESP32#2225] [General]
I can't compile the project because a missing Module library for the ESP32 S2.

**Comment by proddy** [comment]
I think the S2 is 4MB and no additional PSRAM? You need min 8MB and PSRAM to run the Modbus module as it consumes a lot of memory.

**Reply by MichaelDvP** [reply]
He means that there is no esp32s2 version in https://github.com/emsesp/EMS-ESP-Modules

**Reply by proddy** [reply]
I know - by design.

**Comment by math322** [comment]
Strange, it works with Version 3.6.4 and flash size is 4mb and psram is 2mb.

**Reply by MichaelDvP** [reply]
@math322  You can skip the modules: remove the lib_dep in platformio.ini and copy the moduleLibrary.cpp/h files from lib_standalone to the src folder.

**Reply by math322** [reply]
Thanks, it works.

**Reply by proddy** [reply]
I've added an s2 to the modules so it should also work now natively
